packages/config/src/env/main.js

```diff
@@ -96,6 +97,23 @@ const getGeneralEnv = async function ({
   })
 }
 
+const getDeployUrls = function ({
+  siteInfo: { name = DEFAULT_SITE_NAME, ssl_url: sslUrl, build_settings: { repo_url: REPOSITORY_URL } = {} },
+  branch,
+  deployId,
+}) {
+  return {
+    URL: sslUrl,
+    REPOSITORY_URL,
+    DEPLOY_PRIME_URL: `https://${branch}--${name}${NETLIFY_DEFAULT_DOMAIN}`,
```

Comment:
Where do we get `branch` and `deployId` from?

`branch` might not be available if running from a non `git` folder

Reply:
`deployId` is taken from:

https://github.com/netlify/build/blob/e118a5c6f71189662d0163af2576ccda85c2e813/packages/config/src/options/main.js#L47

Before this PR, it was set either the `@netlify/config` flag `deployId` (which we use [in production](https://github.com/netlify/buildbot/blob/cc2414d03cab24be735f490c52fb835c70eede76/script/run-build.sh#L79)) or the `DEPLOY_ID` environment variable. With this PR, it also defaults to `0` for CLI and programmatic builds.

---

`branch` is taken from:

https://github.com/netlify/build/blob/main/packages/config/src/options/branch.js#L3-L9

It has a default value when used in a non-git environment.